src/block.js

```diff
@@ -89,17 +92,21 @@ const get = (source, path) => {
 }
 
 /**
- * @template T
+ * @template {unknown} T - Logical type of the data encoded in the block
+ * @template {number} C - multicodec code corresponding to codec used to encode the block
+ * @template {number} A - multicodec code corresponding to the hashing algorithm used in CID creation.
+ * @template {API.Version} V - CID version
+ * @implements {API.BlockView<T, C, A, V>}
  */
 class Block {
   /**
    * @param {Object} options
-   * @param {CID} options.cid
-   * @param {ByteView<T>} options.bytes
+   * @param {CID<T, C, A, V>} options.cid
```

Comment:
can we fully `Link` the `Block` cass?

Reply:
We could but in that case it would be a breaking change & I don’t think it’s worth it. We could make this  specific argument type into `Link` however. But then again, it is more of an internal function, user should use factory instead.